### ISSUE TITLE
Change behavior when number of samples of either train or validation is too small.

### ISSUE THREAD

**Issue by remtav** [NRCan/geo-deep-learning#135]
Suggested modifications:

Before training:
- assert length of train and val samples in hdf5 is ­equal or larger than "batch size"
- if val samples fails assert, expected behavior would be to continue to test loop. Not purpose in training with no val loop.
- elif train samples fails assert: warn user and continue (see below)

During training, if not enough train_samples to perform train loop, expected behavior of gdl is to:
- skip train loop and go on to val loop only once (for example, num_epochs could be overwritten to 0 after val loop)

Currently, loss is set None and raises an error.

NB: This bugfix could also be seen as a feature: useful for running only validation loop (if num_trn_samples is manually set to 0).

**Comment by victorlazio109** [comment]
While I took a look at how to implement the modifications it became apparent that the train_*py and config*yaml structure may need to be redefined to make such functionalities more explicit. A simple hotfix has been proposed in the linked pull request. Afterward, we can discuss with the broad team on how to implement the suggested modifications in the best possible way.
